aspnetcore/signalr/hub-filters.md

```diff
@@ -0,0 +1,196 @@
+    public async ValueTask<object> InvokeMethodAsync(HubInvocationContext invocationContext, Func<HubInvocationContext, ValueTask<object>> next)
+    {
+        var languageFilter = (LanguageFilterAttribute)Attribute.GetCustomAttribute(invocationContext.HubMethod, typeof(LanguageFilterAttribute));
+        if (languageFilter != null &&
+            invocationContext.HubMethodArguments.Count > languageFilter.FilterArgument &&
+            invocationContext.HubMethodArguments[languageFilter.FilterArgument] is string str)
+        {
+            foreach (var bannedPhrase in bannedPhrases)
+            {
+                str = str.Replace(bannedPhrase, "***");
+            }
+
+            arguments = invocationContext.HubMethodArguments.ToArray();
+            arguments[languageFilter.FilterArgument] = str;
+            invocationContext = new HubInvocationContext(invocationContext.Context,
+                invocationContext.ServiceProvider,
+                invocationContext.Hub,
+                invocationContext.HubMethod,
+                arguments);
+        }
+
+        return await next(invocationContext);
+    }
+}
+```
+
+Finally, we need to register the hub filter. To avoid reinitializing the banned phrases list for every invocation, we register the hub filter as a singleton.
+
+```csharp
+public void ConfigureServices(IServiceCollection services)
+{
+    services.AddSignalR(hubOptions =>
+    {
+        hubOptions.AddFilter<LanguageFilter>();
+    });
+
+    services.AddSingleton<LanguageFilter>();
+}
+```
+
+## The HubInvocationContext object
+
+The `HubInvocationContext` contains information for the current hub method invocation.
+
+| Property | Description | Type |
+| ------ | ------ | ----------- |
+| `Context ` | The `HubCallerContext` contains information about the connection. | `HubCallerContext` |
+| `Hub` | The instance of the Hub being used for this hub method invocation. | `Hub` |
+| `HubMethodName` | The name of the hub method being invoked. | `string` |
+| `HubMethodArguments` | The list of arguments being passed to the hub method. | `IReadOnlyList<string>` |
+| `ServiceProvider` | The scoped service provider for this hub method invocation. | `IServiceProvider` |
+| `HubMethod` | The hub method information. | `MethodInfo` |
+
+## The HubLifetimeContext object
+
+The `HubLifetimeContext` contains information for the `OnConnectedAsync` and `OnDisconnectedAsync` hub methods.
+
+| Property | Description | Type |
+| ------ | ------ | ----------- |
+| `Context ` | The `HubCallerContext` contains information about the connection. | `HubCallerContext` |
+| `Hub` | The instance of the Hub being used for this hub method invocation. | `Hub` |
+| `ServiceProvider` | The scoped service provider for this hub method invocation. | `IServiceProvider` |
+
+## Authorization and filters
+
+[Authorize attributes on hub methods](xref:signalr/authn-and-authz#use-authorization-handlers-to-customize-hub-method-authorization) run before hub filters.
```

Comment:
Any suggestions for this section? It is info we'd like to include but it feels odd having it at the end and such a small section.

Reply:
I think this is fine.

Reply:
I also think it's fine.